cql3/selection/selector_factories.hh

```diff
@@ -121,6 +123,36 @@ public:
         return _factories[0]->is_count_selector_factory();
     }
 
+    bool does_reduction() const {
+        if (_factories.size() != 1) {
+            return false;
+        }
+
+        return _factories[0]->is_reducible_selector_factory() && _factories[0]->contains_only_simple_arguments();
+    }
+
+    // std::vector<std::pair<query::forward_request::reduction_type, >>
+    std::pair<std::vector<query::forward_request::reduction_type>, std::vector<query::forward_request::aggregation_info>>
```

Comment:
It's better to use a struct instead of std::pair, so the names guide readers about what's going on.

Reply:
changed to struct